lib/stub_alloc_pages.c

```diff
@@ -42,6 +42,15 @@
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 
+/* Expose the address of the bigarray (in other words, leak the pointer).
+  */
+CAMLprim value
+caml_address(value t)
+{
+  long data = (long)Caml_ba_data_val(t) ;
+  return Val_long(data);
```

Comment:
Also names with the `caml_` prefix should probably be left for things released with the compiler.